imageproxy.go

```diff
@@ -214,16 +213,14 @@ func (p *Proxy) serveImage(w http.ResponseWriter, r *http.Request) {
 	if err != nil {
 		msg := fmt.Sprintf("error fetching remote image: %v", err)
 		p.log(msg)
-		r, _ := regexp.Compile("address matches a denied host$")
 
-		if r.MatchString(err.Error()) {
+		if v, ok := err.(*url.Error); ok && v.Err.Error() == "address matches a denied host" {
```

Comment:
The error returned from `p.Client.Do(actualReq)` could be a `url.Error` which wraps the original error, by taking care of that case we can avoid checking for the substring via a Regexp.

Reply:
Oh, nice! Thanks @intrip! This was all my ignorant doing so that change seems perfect to me. Also, TIL!
```go
v, ok := err.(*url.Error);
```
I didn't know about [this](https://go.dev/ref/spec#Type_assertions) at all! 🙇

Reply:
I'm pleased I can give back a little from all the learning I got from you ❤️